app/main.py

```diff
@@ -1 +1,41 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self, name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+
+        Animal.alive.append(self)
+
+    def take_damage(self) -> None:
+        self.health -= 50
+        if self.health <= 0:
+            Animal.alive.remove(self)
+        return None
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}, Health: {self.health}, Hidden: {self.hidden}}}"
+
+    @classmethod
+    def __str__() -> str:
```

Comment:
The `__str__` method should not be a class method. It should be an instance method to provide a string representation of an instance of the class. Consider removing the `@classmethod` decorator and adjusting the method accordingly.